libs/dh/admin/data-access-api/src/lib/dh-admin-create-user-role-management-data-access-api.store.ts

```diff
@@ -57,16 +54,16 @@ export class DhAdminCreateUserRoleManagementDataAccessApiStore extends Component
     super(initialState);
   }
 
-  private readonly getSelectablePermissions = this.effect(
-    (trigger$: Observable<void>) =>
+  public readonly getSelectablePermissions = this.effect(
+    (trigger$: Observable<EicFunction>) =>
       trigger$.pipe(
-        switchMap(() =>
+        switchMap((eicFunction) =>
           this.httpClientUserRole
-            .v1MarketParticipantUserRolePermissionsGet()
+            .v1MarketParticipantUserRolePermissionsGet(eicFunction)
             .pipe(
               tapResponse(
                 (response) => {
-                  this.updatePermissions(response);
+                  this.updatePermissions(response ?? []);
```

Comment:
question: can the `response` here be other than an array?

Reply:
It shouldn't ever be, but in testing it did return null, so until i have had time to investigate i'd like to leave this in.